lua/lspsaga/diagnostic.lua

```diff
@@ -15,6 +15,45 @@ M.highlights = {
   [vim.diagnostic.severity.HINT] = "DiagnosticFloatingHint",
 }
 
+local get_line_diagnostics = function(lnum, bufnr)
+  return function()
+    local buf = bufnr or vim.api.nvim_get_current_buf()
+    local line_num = lnum or (vim.api.nvim_win_get_cursor(0)[1] - 1)
+
+    return vim.diagnostic.get(buf, { lnum = line_num  })
+  end
+end
+
+local format_message = function(diagnostic)
+    local message = string.gsub(config.diagnostic_message_format, '%%m', diagnostic.message)
+
+    if diagnostic.user_data and diagnostic.user_data.lsp and diagnostic.user_data.lsp.code then
+        message = string.gsub(message, '%%c', '[' .. diagnostic.user_data.lsp.code .. ']')
+    else
+        message = string.gsub(message, '%%c', '')
+    end
+
+    return vim.fn.trim(message)
+end
+
+M.yank_line_messages = function(opts, lnum, bufnr)
+  local diagnostics = get_line_diagnostics(lnum, bufnr)()
+  local messages = {}
+  for _, diagnostic in ipairs(diagnostics) do
+    table.insert(messages, format_message(diagnostic))
+  end
+  if #messages == 0 then
+    print('No diagnostics found')
+    return
+  end
+  local reg = opts or '"' -- yank to " by default
```

Comment:
```suggestion
  local reg = opts or '+' -- yank to " by default
```
This is the only thing I'm considering to be default right now. usually users would like to copy to + to send to someone or post some where no?

Reply:
Done.